query-engine/query-engine-node-api/src/node_api/functions.rs

```diff
@@ -103,3 +103,13 @@ pub fn get_config(ctx: CallContext) -> napi::Result<JsUnknown> {
     ctx.env.adjust_external_memory(s.len() as i64)?;
     ctx.env.to_js_value(&serialized)
 }
+
+#[js_function(0)]
+pub fn debug_panic(ctx: CallContext<'_>) -> napi::Result<JsUndefined> {
+    let user_facing = user_facing_errors::Error::from_panic_payload(Box::new("Debug panic"));
+    let message = serde_json::to_string(&user_facing).unwrap();
+
+    Err(napi::Error::from_reason(message))?;
```

Comment:
Why a faked panic instead of an actual panic?

Reply:
https://github.com/prisma/prisma-engines/pull/2201

Reply:
Yes, this simulates the panic handling from the HTTP API, but it probably doesn't match what happens on panic with the node API library.

Reply:
I think it's misleading, and there should be different tests on the TS side. This is not testing anything valuable otherwise imo.

Reply:
It never tested anything valuable...